test/SILOptimizer/return.swift

```diff
@@ -9,6 +9,11 @@ func singleBlock2() -> Int {
   y += 1
 } // expected-error {{missing return in a function expected to return 'Int'}}
 
+enum NoCasesButNotNever {}
+
+func diagnoseNoCaseEnumMissingReturn() -> NoCasesButNotNever {
```

Comment:
Note that this is meant to be an error even with a 'Never' return. a Never-returning function should end with a call to some other Never-returning function, or the return should be unreachable.